src/components/LinePlotContainer/LinePlotContainer.tsx

```diff
@@ -3,57 +3,56 @@ import { Grid, CardContent, Card, ButtonGroup, Button, Paper } from '@material-u
 
 import ArrowForwardIosIcon from '@material-ui/icons/ArrowForwardIos'
 import LinePlot from '@components/LinePlot/LinePlot'
-import { data as mockData } from './mockData'
 import { colors } from '@static/theme'
 
 import useStyles from './style'
 interface Data {
   id: string
-  data: Array<{ x: number; y: number }>
+  points: Array<{ x: number; y: number }>
 }
-
-export const LinePlotContainer: React.FC = () => {
+interface IProp {
+  data: Data[]
+  status: boolean
+}
+export const LinePlotContainer: React.FC<IProp> = ({ data, status }) => {
   const classes = useStyles()
-  const [menuOption, setMenuOption] = React.useState('Volument')
+  const [menuOption, setMenuOption] = React.useState('Volume')
   const [activeButtonTime, setActiveButtonTime] = React.useState('Y')
-  const [data, setData] = React.useState<Data>(mockData[0])
-  const [dataPlot, setDataPlot] = React.useState<Array<{ x: number; y: number }>>(
-    mockData[0].data.sort((a: { x: number; y: number }, b: { x: number; y: number }) => {
-      return a.x - b.x
-    })
-  )
+  const [dataTmp, setDataTmp] = React.useState<Data>(data[0])
+  const [dataPlot, setDataPlot] = React.useState<Array<{ x: number; y: number }>>(data[0].points)
+  // const [status, setStatus] = React.useState(false)
   const changeData = (name: string) => {
-    const value = mockData.findIndex(element => element.id === name)
+    const value = data.findIndex(element => element.id === name)
 
-    setData(mockData[value])
+    setDataTmp(data[value])
   }
 
   const sortData = (type: string) => {
     const timestamp = Date.now()
     if (type === 'D') {
       setDataPlot(
-        data.data.filter(element => {
+        dataTmp.points.filter(element => {
           return element.x > timestamp - 86400000
```

Comment:
I think, that to not duplicate so much code you can use ifs/switch to calcucate how many ms you have to subtract in filter, and then set dataTmp to filtered data